server/src/main/java/org/elasticsearch/common/xcontent/XContentHelper.java

```diff
@@ -618,4 +615,13 @@ public static XContentParser mapToXContentParser(XContentParserConfiguration con
             throw new ElasticsearchGenerationException("Failed to generate [" + source + "]", e);
         }
     }
+
+    /**
+     * Extract a java type from an {@link XContentParser}.
+     * @param <T> the type to extract
+     */
+    @FunctionalInterface
+    public interface XContentTypeExtractor<T> {
```

Comment:
There is a `CheckedFunction` interface you can use instead...

Reply:
Changed.